sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1115,14 +1168,23 @@ def point_cflexure(self):
         >>> b.point_cflexure()
         [10/3]
         """
+        #Removes the singularity functions of order < 0 from the bending moment equation used in this method
+        non_singular_bending_moment = sum(arg for arg in self.bending_moment().args if not arg.args[1].args[2] < 0)
```

Comment:
@mvg2002 , wouldn't it make sense if the singularity function would never return $\infty$?

Reply:
This has to do with the definition of negative order singularity functions, take for example $< x - a > ^ {-1}$. This function can be in the bending moment equation as result of a rotation hinge at $x = a$. 

The method `point_cflexure` searches for points where the bending moment is equal to 0. As there is a hinge located at  $x = a$ the bending moment should be 0 at that point and the method should return $x = a$ as a point where the bending moment is 0. 

But when $< x - a > ^ {-1}$ is still present in the bending moment equation, evaluating it at $x = a$ would return $\infty$. The method would not give $x = a$ as a point where the bending moment is 0. That is why for this calculation singularity functions of negative order are removed, they don't do anything except set the bending moment to infinity in the hinges. 

The singularity function should return $\infty$ when $x = a$, as follows from their mathematical definition. This should not be changed in SymPy. There might be other people using the singularity functions and they expect them to return $\infty$ when $x = a$

Reply:
I agree that we shouldn't change it everywhere, but wouldn't the mechanics application benefit from a definition without $\infty$?

Reply:
@Tom-van-Woudenberg, For mechanical applications of the singularity function returning $\infty$ is never needed and can only complicate things. I agree that the mechanics application would benifit from a definition without $\infty$.

Reply:
@moorepants, would there be a place in SymPy for mechanics-specific singularity functions?